topic9/homework2/Voter.sol

```diff
@@ -0,0 +1,121 @@
+pragma solidity 0.4.23;
+
+library VotingLib {
+    using SafeMath for uint;
+
+    struct VotingProposal {
+        address addr;
+        uint value;
+        uint votesFor;
+        uint votesAgainst;
+        uint decidingVotes;
+        mapping (address=>bool) voted;
+        bool finished;
+        bool accepted;
+    }
+
+    function createVoting(address _proposal, uint _value, uint _decidingVotes) {
+        return VotingProposal({value: _value, votesFor: 0, votesAgainst: 0, finished: false, accepted: false, addr: _proposal, votes: 0, decidingVotes: _decidingVotes});
+    }
+
+    function vote(Voting storage self, uint importance, bool forOrAgainst) {
+        require(!self.finished);
+        require(!self.voted[msg.sender]);
+
+        self.voted[msg.sender] = true;
+
+        if (forOrAgainst) {
+            self.votesFor = self.votesFor.add(importance);
+
+            if (self.votesFor >= self.decidingVotes) {
+                self.finished = true;
+                self.accepted = true;
+            }
+        } else {
+            self.votesAgainst = self.votesAgainst.add(importance);
+
+            if (self.votesAgainst >= self.decidingVotes) {
+                self.finished = true;
+                self.accepted = false;
+            }
+        }
+    }
+}
+
+
+contract Voter is Ownable {
+    using VotingLib for VotingLib.VotingProposal;
+    using SafeMath for uint;
+
+    event LogVotingStart(address indexed addr, uint value);
+    event LogVotingSuccessful(address indexed addr, uint value);
+    event LogWithdrawal(address indexed addr);
+
+    function () public payable {}
+
+    struct Member {
+        bool isMember;
```

Comment:
Is there a use case where we can have 0 importance of a member? If not - do we need to keep this `isMember` boolean variable?